tests/test_causal_graph.py

```diff
@@ -538,6 +538,14 @@ def test_adjacency_consistency(self):
 
         self.assertEqual(CausalGraph.from_adjacency_matrix(*cg.to_numpy()), cg)
 
+    def test_copy(self):
+        # test copy preserves the correct class
+        ts_cg = TimeSeriesCausalGraph()
+        ts_cg.add_edge('a', 'b')
+
+        ts_cg_copy = ts_cg.copy()
+        self.assertEqual(type(ts_cg_copy), type(ts_cg))
```

Comment:
We know the type of `ts_cg`. Would change to `self.assertIsInstance(ts_cg_copy, TimeSeriesCausalGraph)` then would add an equality check (using the deep one?) that `ts_cg` and `ts_cg_copy` are equal.